### PR TITLE
Convert the data migration to a Django migration

### PR DESCRIPTION
**Pull request by nedbat**

### CODE REVIEW

**Comment by nedbat** [comment]
@macdiesel @doctoryes @symbolist @muzaffaryousaf look ok?

File: edxval/migrations/0002_data__default_profiles.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+def create_default_profiles(apps, schema_editor):
+    Profile = apps.get_model("edxval", "Profile")
+    for kwargs in DEFAULT_PROFILES:
+        Profile.objects.get_or_create(**kwargs)
+
+
+def delete_default_profiles(apps, schema_editor):
+    Profile = apps.get_model("edxval", "Profile")
+    for kwargs in DEFAULT_PROFILES:
+        Profile.objects.get_or_create(**kwargs).delete()
```
Comment (doctoryes) [inline edxval/migrations/0002_data__default_profiles.py]: If you `create()` and `delete()` in the same filter, does it do both in the DB? Or is it smart enough to do nothing?

I think it's fine either way - data migration is a one-time, out-of-band thing.

**Comment by doctoryes** [comment]
:+1:

File: edxval/migrations/0002_data__default_profiles.py (hunk 1 above)
Comment (nedbat) [inline edxval/migrations/0002_data__default_profiles.py]: Oh, oops: this won't work: get_or_create returns a pair.... I'll fix it.

**Comment by nedbat** [comment]
@doctoryes thanks for the skepticism, there were a bunch of things wrong with this migration!
Look again?

File: edxval/migrations/0002_data__default_profiles.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+def create_default_profiles(apps, schema_editor):
+    Profile = apps.get_model("edxval", "Profile")
+    for profile in DEFAULT_PROFILES:
+        Profile.objects.get_or_create(profile_name=profile)
+
+
+def delete_default_profiles(apps, schema_editor):
+    Profile = apps.get_model("edxval", "Profile")
+    for profile in DEFAULT_PROFILES:
+        obj, _ = Profile.objects.get_or_create(profile_name=profile)
```
Comment (muzaffaryousaf) [inline edxval/migrations/0002_data__default_profiles.py]: Why get_or_create ? We need to delete them all. `Profile.objects.filter(profile_name=profile).delete()`

**Comment by muzaffaryousaf** [comment]
@nedbat We already have this data migration in it's initial file. Please see here https://github.com/edx/edx-val/blob/ammar/upgrade-django-1.8/edxval/migrations/0001_initial.py#L9

**Comment by muzaffaryousaf** [comment]
@nedbat We have to remove the data migration from initial file.

**Comment by muzaffaryousaf** [comment]
:+1:

**Comment by doctoryes** [comment]
:+1:

**Pull request merged**